cmd/juju/application/deploy.go

```diff
@@ -387,80 +387,86 @@ type DeployCommand struct {
 }
 
 const deployDoc = `
-<charm or bundle> can be a charm/bundle URL, or an unambiguously condensed
-form of it; assuming a current series of "trusty", the following forms will be
-accepted:
+A charm can be referred to by its simple name or by specifying a series (if
+available):
 
-For cs:trusty/mysql
-  mysql
-  trusty/mysql
+  juju deploy postgresql
+  juju deploy xenial/postgresql
+  juju deploy cs:xenial/postgresql
+  juju deploy postgresql --series xenial
 
-For cs:~user/trusty/mysql
-  ~user/mysql
+The second and third forms are using charm URLs.
```

Comment:
What are the first and fourth forms?  Technically those commands will use the charm store.  Can we have a charm URL not from the charm store?  I don't know the answer to that.  And maybe isn't necessary for this update?

Reply:
Yeah, looks like all those forms are charm store URLs.  Trying to deploy something ambiguous (local charm is in the pwd) gives you a warning to be more specific:
```
ubuntu@b413deb34dfe:~/charms/xenial$ juju deploy pig
ERROR The charm or bundle "pig" is ambiguous.
To deploy a local charm or bundle, run `juju deploy ./pig`.
To deploy a charm or bundle from the store, run `juju deploy cs:pig`.
```

I think we could drop this sentence about 2nd/3rd forms, or note that all forms will communicate with the charm store.  The warning is pretty clear, and deploying local charms is covered below.

Reply:
I'll drop the sentence. I just felt the need to explain "charm URL" somehow.